File: coursera/kotlin-for-java-developers/week1/src/main/java/extensions/extensions/Extensions.java
package extensions.extensions;

import kotlin.text.StringsKt;

public class Extensions {
    public static void main(String[] args) {
        System.out.println(ExtensionsKt.lastChar("abc"));
        assert ExtensionsKt.lastChar("abc") == 'c';
        System.out.println(StringsKt.last("abc"));
        assert StringsKt.last("abc") == 'c';
        System.out.println(ExtensionsKt.repeat("abc", 5));
        assert ExtensionsKt.repeat("abc", 5).equals("abcabcabcabcabc");
    }
}

File: coursera/kotlin-for-java-developers/week2/src/main/kotlin/extensions/extensions/Extensions.kt
package extensions.extensions

fun main() {
    println("${"test".last()}")
    println("${"test".lastChar()}")
    println("test" repeat 5)
}

fun String.lastChar() = get(length - 1)

infix fun String.repeat(n: Int): String {
    val sb = StringBuilder(n * length)
    repeat(n) { sb.append(this) }
    return sb.toString()
}